File: src/com/leetcode/Easy/CountTheNumberOfConsistentStrings.java
package com.leetcode.Easy;

// 1684. Count the Number of Consistent Strings
public class CountTheNumberOfConsistentStrings {
    public int countConsistentStrings(String allowed, String[] words) {
        int count = 0;
        for (String word : words) {
            if (containsDistinct(word, allowed))
                count++;
        }
        return count;
    }

    private boolean containsDistinct(String word, String allowed) {
        for (char ch : word.toCharArray()) {
            if (!allowed.contains(String.valueOf(ch)))
                return false;
        }
        return true;
    }
}

File: JavaScript/Easy/CountTheNumberOfConsistentStrings.js
/**
 * 1684. Count the Number of Consistent Strings
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
const countConsistentStrings = function (allowed, words) {
  let count = 0;
  let allowedSet = new Set(allowed);
  for (const word of words) {
    let isConsistent = true;
    for (const char of word) {
      if (!allowedSet.has(char)) {
        isConsistent = false;
        break;
      }
    }

    if (isConsistent)
      count++;
  }
  return count;
};